src/internal/ReplaySubject.ts

```diff
@@ -80,6 +96,15 @@ export class ReplaySubject<T> extends Subject<T> {
       subscriber.next(copy[i] as T);
     }
 
+    // Emit values that are synchronously pushed
+    for (let i = 0; i < syncValues.length && !subscriber.closed; i++) {
+      subscriber.next(syncValues[i]);
+    }
+
+    // Swap subscription to the original one
+    syncSubscription.unsubscribe();
+    const subscription = this._innerSubscribe(subscriber);
+
     this._checkFinalizedStatuses(subscriber);
```

Comment:
we only need to execute `_innerSubscribe` if `_checkFinalizedStatuses` did not emit complete or error:
```ts
if (this._checkFinalizedStatuses(subscriber)) { // needs a small change in `Subject` to return a bool
  return subscriber;
}

return this._innerSubscribe(subscriber);
```

Reply:
Same thing, I think it's a better design to be transparent to the underlying Subject. Subject expects `this._innerSubscribe` to be called with the subscriber who subscribed to the subject before calling `this._checkFinalizedStatuses`.

CheckFinalizedStatuses will cleanup everything if needed - This keeps the separation of concerns intact.